### PR TITLE
Make an example of how to run Qtile on Fedora. (Fix #1417)

### LINKED ISSUE #1417: docs: how to install Qtile from source in detail?
Hi,

Installing Qtile via a distro's package manager is a no-brainer.

But, If I want to try the source of master, I'm not sure how to do that exactly. There is a [section](http://docs.qtile.org/en/latest/manual/install/index.html#installing-from-source) on the site about how to install dependent packages, but it doesn't mention how to install Qtile itself actually.

My thought is to:
1. create a system-wide virtualenv
2. copy the source to somewhere, e.g. `/usr/local/bin/`
3. create a new file `/usr/share/xsessions/qtile-venv.desktop`

Am I right? How do you do that?

### PR DESCRIPTION
**Pull request by whatacold**
Add an example about how to run Qtile within a virtualenv on Fedora, this fixes #1417 . Correct me if anything is incorrect, be it English grammar or the content.

### CODE REVIEW

File: docs/manual/install/index.rst [hunk 1]
```diff
@@ -45,14 +45,14 @@ cairocffi
...
-install cairocffi that xcffib has been installed, otherwise the needed
+the underlying library used by the binding.  You should **be sure before you
+install cairocffi that xcffib has been installed**, otherwise the needed
 cairo-xcb bindings will not be built.  Once you've got the dependencies
 installed, you can use the latest version on PyPI:
 
 .. code-block:: bash
 
-    pip install cairocffi
+    pip install --no-cache-dir cairocffi
```
Comment (flacjacket) [inline docs/manual/install/index.rst]: This is nice, we may want to add this to the error message that we print if cairocffi is not properly installed.

File: docs/manual/install/index.rst [hunk 2]
```diff
@@ -92,3 +92,36 @@ Stable versions of Qtile can be installed from PyPI:
 As long as the necessary libraries are in place, this can be done at any point,
 however, it is recommended that you first install xcffib to ensure the
 cairo-xcb bindings are built (see above).
+
+To recap, let's take it as an example on Fedora distro to run qtile within a virtualenv,
```
Comment (flacjacket) [inline docs/manual/install/index.rst]: `let's` to `lets`

File: docs/manual/install/index.rst [hunk 3]
```diff
@@ -92,3 +92,36 @@ Stable versions of Qtile can be installed from PyPI:
...
+the virtualenv is optional (assume that the git repo is cloned at ``~/local/qtile/`` of user ``foo``):
+
+#. (optional) Create a virtualenv ``~/local/qtile/qtile-env/``, and install the dependencies there.
+#. (optional) Create a glue shell to take advantage of the virtualenv.
```
Comment (flacjacket) [inline docs/manual/install/index.rst]: Is this the `/home/foo/local/qtile/qtile-venv-entry` file? Is that optional? May be good to note where the output should go.

File: docs/manual/install/index.rst (hunk 1 above)
Comment (whatacold) [reply docs/manual/install/index.rst]: Where would be the best place to print this error message?

Comment (flacjacket) [reply docs/manual/install/index.rst]: Ah, I was thinking in `setup.py`, but it looks like that error message already has this.

**Comment by whatacold** [comment]
@flacjacket, thanks for the review.

I rephrased the steps to make it more clear on how to run it within a virtualenv step by step.

File: docs/manual/install/index.rst [hunk 4]
```diff
@@ -92,3 +92,41 @@ Stable versions of Qtile can be installed from PyPI:
...
 cairo-xcb bindings are built (see above).
+
+The above steps are sufficient to run Qtile directly, but there are some extra works if you want to run it within a virtualenv. Here are the steps on a Fedora system for user ``foo``, it should work on other Linux systems too.
+
+#. Clone the repo as ``~/local/qtile/``.
+
+   .. code-block:: bash
+
+       mkdir -p ~/local/
+       git clone git://github.com/qtile/qtile.git
```
Comment (flacjacket) [inline docs/manual/install/index.rst]: `cd ~/local/`

Comment (whatacold) [reply docs/manual/install/index.rst]: Fixed, thanks!

**Comment by whatacold** [comment]
@flacjacket I would like to do a squashed rebase when you think it's ok to do so, so that the commit will be more relevant.

**Comment by flacjacket** [comment]
This is fine, thanks for the updates!

**Comment by flacjacket** [comment]
Feel free to squash if you would like to, otherwise we can just pull in the commits as-is.

**Comment by whatacold** [comment]
OK, rebased and squashed.

**Comment by flacjacket** [comment]
Sounds good, thanks!

**Pull request merged**